src/reducers/sources.js

```diff
@@ -187,19 +198,20 @@ function updateSource(state: SourcesState, source: Object) {
     ? { ...existingSource, ...source }
     : createSource(source);
 
+  state.sources[source.id] = updatedSource;
+
   const existingUrls = state.urls[source.url];
-  const urls = existingUrls ? [...existingUrls, source.id] : [source.id];
+  state.urls[source.url] = existingUrls
```

Comment:
Do we need to take into account origin vs. generated here?  Since a sourcemap can have the same URL?